tests/helpers/test_script.py

```diff
@@ -56,6 +56,39 @@ def record_event(event):
         assert calls[0].data.get('hello') == 'world'
         assert not script_obj.can_cancel
 
+    def test_firing_event_template(self):
+        """Test the firing of events."""
+        event = 'test_event'
+        calls = []
+
+        @callback
+        def record_event(event):
+            """Add recorded event to set."""
+            calls.append(event)
+
+        self.hass.bus.listen(event, record_event)
+
+        script_obj = script.Script(self.hass, cv.SCRIPT_SCHEMA({
+            'event': event,
+            'event_data_template': {
+                'hello': """
+                    {% if True %}
```

Comment:
Please use a passed in variable.

Reply:
Again I copied this from service data template, but it bothered me that there was no variable in that test either, so I added it in both places.